src/main/java/games/strategy/triplea/delegate/MoveValidator.java

```diff
@@ -1057,19 +1057,17 @@ private static MoveValidationResult validateTransport(final boolean isNonCombat,
           }
         }
       }
-    } // end if end is water
+    }
     if (route.isLoad()) {
       if (!isEditMode && !route.hasExactlyOneStep() && nonParatroopersPresent(player, landAndAir)) {
         return result.setErrorReturnResult("Units cannot move before loading onto transports");
       }
       final Predicate<Unit> enemyNonSubmerged = Matches.enemyUnit(player, data).and(Matches.unitIsSubmerged().negate());
-      if (route.getEnd().getUnits().anyMatch(enemyNonSubmerged) && nonParatroopersPresent(player, landAndAir)) {
-        if (!onlyIgnoredUnitsOnPath(route, player, data, false)) {
-          if (!AbstractMoveDelegate.getBattleTracker(data).didAllThesePlayersJustGoToWarThisTurn(player,
+      if (!Properties.getUnitsCanLoadInHostileSeaZones(data) && route.getEnd().getUnits().anyMatch(enemyNonSubmerged)
```

Comment:
Improved, still heavy to read. There may be some gains by reducing the transitive calls, making the objects a bit smarter, noting this one good example:
```
route.getEnd().getUnits().anyMatch(enemyNonSubmerged)
```
```
route.endOfRouteHasNonSubmergedUnits();
```

That would help for one of these conditions.  This is a bit of a tough block to simplify. Extracting a few or all conditions to variables to give them even more naming _maybe_ could help.

Reply:
Agree from a readability standpoint. The challenge is the huge number of methods that would need added to wrap all of these hard to read calls. In the end, it would be more code to maintain in order to improve readability. Hard to say whether or not its worth it.